Don't fail if fresh_inductor_cache fails to clean up its tmp dir.

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #145513

Summary: I see we have a test failure due to an error removing the tmp dir: https://github.com/pytorch/pytorch/issues/141761. Seems like we should not raise an exception for this case in general. Also, let's clean up the exception handling related to windows. The comment makes it sound like we want to specifically ignore failures cleaning up, but the current impl is swallowing all exceptions.

Fixes #141761

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

masnesral committed:
Don't fail if fresh_inductor_cache fails to clean up its tmp dir.

Summary: I see we have a test failure due to an error removing the tmp dir: https://github.com/pytorch/pytorch/issues/141761. Seems like we should not raise an exception for this case in general. Also, let's clean up the exception handling related to windows. The comment makes it sound like we want to specifically ignore failures cleaning up, but the current impl is swallowing all exceptions.

Fixes #141761

[ghstack-poisoned]

eellison reviewed (approved):
nice !

Skylion007 commented on torch/_inductor/utils.py:
```diff
@@ -855,15 +855,16 @@ def fresh_inductor_cache(cache_entries=None, dir=None, delete=True):
                         )
         if delete:
-            shutil.rmtree(inductor_cache_dir)
+            try:
+                shutil.rmtree(inductor_cache_dir)
+            except OSError:
+                # Let's not fail if we can't clean up the temp dir. Also note that for
+                # Windows, we can't delete the loaded modules because the module binaries
+                # are open.
+                log.warning("Failed to remove temporary cache dir at %s", inductor_cache_dir)
```
```suggestion
           shutil.rmtree(inductor_cache_dir, ignore_errors=True, on_error=lambda: log.warning("Failed to remove temporary cache dir at %s", inductor_cache_dir))
```

Skylion007 replied:
shutl hsas an onerror callback, let's just use that

masnesral committed:
Update on "Don't fail if fresh_inductor_cache fails to clean up its tmp dir."


Summary: I see we have a test failure due to an error removing the tmp dir: https://github.com/pytorch/pytorch/issues/141761. Seems like we should not raise an exception for this case in general. Also, let's clean up the exception handling related to windows. The comment makes it sound like we want to specifically ignore failures cleaning up, but the current impl is swallowing all exceptions.

Fixes #141761

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/utils.py:
```diff
@@ -855,15 +855,18 @@ def fresh_inductor_cache(cache_entries=None, dir=None, delete=True):
-            """
-            log.warning("on error, temporary cache dir kept at %s", inductor_cache_dir)
-            raise
+        log.warning("on error, temporary cache dir kept at %s", inductor_cache_dir)
```
```suggestion
        log.warning("on error, temporary cache dir kept at %s", inductor_cache_dir, exc_info=True)
```
? want to include error info?

masnesral replied:
@Skylion007 I wouldn't think we'd want to here. The exception is unrelated to the rmtree and is being raised. I think the intention here is just to let you know where the inductor artifacts are located for debugging purposes?

Skylion007 commented on torch/_inductor/utils.py:
```diff
@@ -855,15 +855,18 @@ def fresh_inductor_cache(cache_entries=None, dir=None, delete=True):
+                # Windows, we can't delete the loaded modules because the module binaries
+                # are open.
+                onerror=lambda func, path, exc_info: log.warning(
+                    "Failed to remove temporary cache dir at %s", inductor_cache_dir
```
```suggestion
                    "Failed to remove temporary cache dir at %s", inductor_cache_dir, exc_info=exc_info
```

masnesral committed:
Update on "Don't fail if fresh_inductor_cache fails to clean up its tmp dir."


Summary: I see we have a test failure due to an error removing the tmp dir: https://github.com/pytorch/pytorch/issues/141761. Seems like we should not raise an exception for this case in general. Also, let's clean up the exception handling related to windows. The comment makes it sound like we want to specifically ignore failures cleaning up, but the current impl is swallowing all exceptions.

Fixes #141761

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge
